sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1188,6 +1188,12 @@ class Dataset[T] private[sql](
   def col(colName: String): Column = colName match {
     case "*" =>
       Column(ResolvedStar(queryExecution.analyzed.output))
+    case ParserUtils.escapedIdentifier(columnNameRegex)
```

Comment:
I am not sure if I understand what you said. no matter what, the colName is string, you need to figure out it is a regex or not.